File: app/src/main/java/mega/privacy/android/app/modalbottomsheet/MeetingBottomSheetDialogFragment.java
package mega.privacy.android.app.modalbottomsheet;

import android.annotation.SuppressLint;
import android.app.Dialog;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;

import androidx.appcompat.app.AppCompatDialogFragment;

import com.google.android.material.bottomsheet.BottomSheetBehavior;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

import java.util.ArrayList;
import java.util.List;

import mega.privacy.android.app.MegaContactAdapter;
import mega.privacy.android.app.R;
import mega.privacy.android.app.components.MarqueeTextView;
import mega.privacy.android.app.components.RoundedImageView;
import mega.privacy.android.app.components.twemoji.EmojiTextView;
import mega.privacy.android.app.lollipop.ManagerActivityLollipop;
import mega.privacy.android.app.lollipop.controllers.ChatController;
import mega.privacy.android.app.lollipop.controllers.ContactController;
import mega.privacy.android.app.utils.ContactUtil;
import nz.mega.sdk.MegaUser;

import static mega.privacy.android.app.utils.AlertsAndWarnings.showOverDiskQuotaPaywallWarning;
import static mega.privacy.android.app.utils.AvatarUtil.setImageAvatar;
import static mega.privacy.android.app.utils.CallUtil.participatingInACall;
import static mega.privacy.android.app.utils.CallUtil.startNewCall;
import static mega.privacy.android.app.utils.ChatUtil.setContactLastGreen;
import static mega.privacy.android.app.utils.ChatUtil.setContactStatus;
import static mega.privacy.android.app.utils.Constants.EMAIL;
import static mega.privacy.android.app.utils.Constants.MAX_WIDTH_BOTTOM_SHEET_DIALOG_LAND;
import static mega.privacy.android.app.utils.Constants.MAX_WIDTH_BOTTOM_SHEET_DIALOG_PORT;
import static mega.privacy.android.app.utils.ContactUtil.getContactDB;
import static mega.privacy.android.app.utils.ContactUtil.getMegaUserNameDB;
import static mega.privacy.android.app.utils.LogUtil.logWarning;
import static mega.privacy.android.app.utils.TextUtil.isTextEmpty;
import static mega.privacy.android.app.utils.Util.dp2px;
import static mega.privacy.android.app.utils.Util.isScreenInPortrait;
import static nz.mega.sdk.MegaApiJava.STORAGE_STATE_PAYWALL;


public class MeetingBottomSheetDialogFragment extends BaseBottomSheetDialogFragment implements View.OnClickListener {

    private ContactController cC;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        cC = new ContactController(context);
    }

    @SuppressLint("RestrictedApi")
    @Override
    public void setupDialog(final Dialog dialog, int style) {
        super.setupDialog(dialog, style);
        contentView = View.inflate(getContext(), R.layout.bottom_sheet_meeting, null);
        mainLinearLayout = contentView.findViewById(R.id.meeting_bottom_sheet);
        items_layout = contentView.findViewById(R.id.meeting_item);
        ImageView startMeeting = contentView.findViewById(R.id.iv_start_meeting);
        ImageView joinMeeting = contentView.findViewById(R.id.iv_join_meeting);
        startMeeting.setOnClickListener(this);
        joinMeeting.setOnClickListener(this);
        dialog.setContentView(contentView);
        setBottomSheetBehavior(HEIGHT_HEADER_LARGE, true);
    }

    @Override
    public void onClick(View v) {

        switch (v.getId()) {
            case R.id.iv_start_meeting:
//                ContactUtil.openContactInfoActivity(context, contact.getMegaUser().getEmail());
                break;

            case R.id.iv_join_meeting:
//                ((ManagerActivityLollipop) context).startOneToOneChat(contact.getMegaUser());
                break;


        }
        setStateBottomSheetBehaviorHidden();
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
    }

}

File: app/src/main/java/mega/privacy/android/app/modalbottomsheet/MeetingBottomSheetDialogFragment.kt
package mega.privacy.android.app.modalbottomsheet

import android.annotation.SuppressLint
import android.app.Dialog
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import com.google.android.material.bottomsheet.BottomSheetBehavior
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import dagger.hilt.android.AndroidEntryPoint
import mega.privacy.android.analytics.Analytics
import mega.privacy.android.app.R
import mega.privacy.android.app.databinding.BottomSheetMeetingSimpleBinding
import mega.privacy.android.app.interfaces.MeetingBottomSheetDialogActionListener
import mega.privacy.mobile.analytics.event.ScheduleMeetingMenuItemEvent

@AndroidEntryPoint
class MeetingBottomSheetDialogFragment : BottomSheetDialogFragment(), View.OnClickListener {

    companion object {
        const val TAG = "MeetingBottomSheetDialog"

        @JvmStatic
        fun newInstance(): MeetingBottomSheetDialogFragment = MeetingBottomSheetDialogFragment()
    }

    private var listener: MeetingBottomSheetDialogActionListener? = null

    /**
     * onScheduleMeeting callback
     */
    var onScheduleMeeting: (() -> Unit)? = null

    @SuppressLint("RestrictedApi")
    override fun setupDialog(dialog: Dialog, style: Int) {
        super.setupDialog(dialog, style)

        val binding =
            BottomSheetMeetingSimpleBinding.inflate(LayoutInflater.from(context), null, false)
        binding.btnScheduleMeeting.setOnClickListener(this)
        binding.btnStartMeeting.setOnClickListener(this)
        binding.btnJoinMeeting.setOnClickListener(this)

        dialog.setContentView(binding.root)
    }

    override fun onStart() {
        super.onStart()
        val dialog = dialog ?: return
        BottomSheetBehavior.from(dialog.findViewById(com.google.android.material.R.id.design_bottom_sheet)).state =
            BottomSheetBehavior.STATE_EXPANDED
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btn_start_meeting -> {
                listener?.onCreateMeeting()
                dismiss()
            }

            R.id.btn_join_meeting -> {
                listener?.onJoinMeeting()
                dismiss()
            }

            R.id.btn_schedule_meeting -> {
                Analytics.tracker.trackEvent(ScheduleMeetingMenuItemEvent)
                onScheduleMeeting?.invoke()
                dismiss()
            }
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        listener = context as MeetingBottomSheetDialogActionListener
    }
}
